src/system/SystemPacketBuffer.cpp

```diff
@@ -530,12 +530,19 @@ PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aRese
 
     CHIP_SYSTEM_FAULT_INJECT(FaultInjection::kFault_PacketBufferNew, return PacketBufferHandle());
 
-    // TODO: Change the max to a lower value
-    if (aAvailableSize > UINT32_MAX || lAllocSize > PacketBuffer::kMaxSizeWithoutReserve || lBlockSize > UINT32_MAX)
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    if (lAllocSize > CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES || lBlockSize > UINT32_MAX)
```

Comment:
should we static-assert that `CHIP_CONFIG_MAX_LARGE_PAYLOAD_SIZE_BYTES >= PacketBuffer::kMaxSizeWithoutReserve` to preserve previous/non-tcp restrictions?